examples/2-handle-prompt/index.js

```diff
@@ -0,0 +1,111 @@
+
+    console.log('Received POST request');
+
+    // get a token to use
+    const tokenForUser = request.headers['x-github-token'];
+    
+    // get the user information with the token        
+    const octokit = new Octokit({ auth: tokenForUser });
+    const user = await octokit.request("GET /user");
+    const userHandle = user.data.login
+    
+    // Collect incoming data chunks to use in the `on("end")` event
+    let body = '';
+    request.on('data', chunk => {
+        body += chunk.toString();
+    });
+
+    // Parse the collected data once the request ends
+    request.on('end', async () => {
+        try {
+            const payloadIsVerified = await verifyRequestByKeyId(
+                request.body,
+                signature,
+                keyId,
+                {
+                  token: process.env.GITHUB_TOKEN,
+                },
+            );
+
+            if (!payloadIsVerified) {
+                console.error('Request verification failed');
+                response.writeHead(401, { 'Content-Type': 'text/plain' });
+                response.end('Request could not be verified');
+                return;
+            }
+
+            // start sending the response back to Copilot
+            console.log('Handling response');
+            // write the acknowledge event to let Copilot know we are handling the request
+            // this will also show the message "Copilot is responding" in the chat
+            response.write(createAckEvent());
+
+            // parse the incoming body as that has the information we need to handle the request / user prompt
+            const payload = parseRequestBody(body);
+            
+            // start writing text in the response
+            const textEvent = createTextEvent(`Hello ${userHandle}, \n\n`);
+            response.write(textEvent);
+            // add new lines to mark the difference between the fixed text and the dynamic text
+            response.write("\n\n");
+
+            // get an authentication token to use
+            const tokenForUser = request.headers['x-github-token'];
+
+            console.log('Calling the GitHub Copilot API with the user prompt');
+            // the prompt to forward to the Copilot API is the last message in the payload
+            const payload_message = payload.messages[payload.messages.length - 1];
+            const result = await prompt(payload_message.content, {
+                system: "you talk like a pirate", // extra instructions for the prompt, the "you are a helpful assistant" is already set in the SDK
+                messages: payload.messages, // we are giving the prompt the existing messages in this chat conversation
+                token: tokenForUser         
+            });
+
+            // write the prompt response back to Copilot
+            // note that this is only send when the entire response from the Copilot API is ready
+            response.write(createTextEvent(result.message.content));
```

Comment:
there is a `prompt.stream` API but we can do a separate example for that if you like?